overlord/ifacestate/handlers.go

```
@@ -327,30 +342,43 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	err = m.repo.Disconnect(plugRef.Snap, plugRef.Name, slotRef.Snap, slotRef.Name)
-	if err != nil {
-		return err
-	}
-
-	plug := m.repo.Plug(plugRef.Snap, plugRef.Name)
-	var plugSnapst snapstate.SnapState
-	if err := snapstate.Get(st, plugRef.Snap, &plugSnapst); err != nil {
-		return err
-	}
-	slot := m.repo.Slot(slotRef.Snap, slotRef.Name)
-	var slotSnapst snapstate.SnapState
-	if err := snapstate.Get(st, slotRef.Snap, &slotSnapst); err != nil {
-		return err
+	var affectedConns []interfaces.ConnRef
+	if plugRef.Snap != "" && plugRef.Name != "" && slotRef.Snap != "" && slotRef.Name != "" {
+		if err := m.repo.Disconnect(plugRef.Snap, plugRef.Name, slotRef.Snap, slotRef.Name); err != nil {
+			return err
+		}
+		affectedConns = []interfaces.ConnRef{{*plugRef, *slotRef}}
+	} else if plugRef.Name != "" {
```

Should we try to explicitly list all the valid cases in the if/else clauses here? In this case for example user could provide plug name *and* slot snap name, hoping that this is going to narrow the effect of disconnect down, but in fact I think it would be silently ignored?